app.js

```diff
@@ -0,0 +1,160 @@
+require("dotenv").config();
+const express = require("express");
+const session = require("express-session");
+const passport = require("passport");
+const LocalStrategy = require("passport-local").Strategy;
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+const bcrypt = require("bcryptjs");
+const MongoDBStore = require("connect-mongodb-session")(session);
+
+//MONGO
+const mongoDb = process.env.MONGO_URI;
+mongoose.connect(mongoDb, {
+  useUnifiedTopology: true,
+  useNewUrlParser: true,
+});
+const db = mongoose.connection;
+db.on("error", console.error.bind(console, "mongo connection error"));
+
+var store = new MongoDBStore({
+  uri: process.env.MONGO_URI,
+  collection: "sessions",
+});
+
+// Catch errors
+store.on("error", function (error) {
+  console.log(error);
+});
+
+const User = mongoose.model(
+  "User",
+  new Schema({
+    username: { type: String, required: true },
+    password: { type: String, required: true },
+  })
+);
+
+const authMiddleware = (req, res, next) => {
+  if (!req.user) {
+    if (!req.session.messages) {
+      req.session.messages = [];
+    }
+    req.session.messages.push("You can't access that page before logon.");
+    res.redirect("/");
+  } else {
+    next();
+  }
+};
+
+const app = express();
+app.set("views", __dirname);
+app.set("view engine", "ejs");
+
+app.use(
+  session({
+    secret: process.env.SESSION_SECRET,
+    resave: false,
+    saveUninitialized: true,
+    store: store,
+  })
+);
+
+app.use(passport.initialize());
+app.use(passport.session());
+app.use(express.urlencoded({ extended: false }));
+
+app.use(function (req, res, next) {
+  res.locals.user = req.user;
+  next();
+});
+
+app.get("/", (req, res) => {
+  let messages = [];
+  if (req.session.messages) {
+    messages = req.session.messages;
+    req.session.messages = [];
+  }
+  res.render("index", { messages });
+});
+
+app.get("/sign-up", (req, res) => {
+  let messages = [];
+  if (req.session.messages) {
+    messages = req.session.messages;
+    req.session.messages = [];
+  }
```

Comment:
There's no issue with having this here; but just want to point out that there's currently nothing that is putting messages into the session for the sign up page. 

If you wanted to, you could add messages to the session if errors occur during signup on line 105:

```js
app.post("/sign-up", async (req, res, next) => {
  try {
    const hashedPassword = await bcrypt.hash(req.body.password, 10);
    await User.create({
      username: req.body.username,
      password: hashedPassword,
    });
    res.redirect("/");
  } catch (err) {
    if (!req.session.messages) {
        req.session.messages = [];
    }
    req.session.messages.push(err.message);
    return next(err);
  }
});
```

Reply:
It is done 👍